### PR TITLE
NAS-132840 / 25.04 / Audit all public `keychaincredential` endpoints

### PR DESCRIPTION
**Pull request by creatorcary**

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]

File: src/middlewared/middlewared/plugins/keychain.py [hunk 1]
```diff
@@ -410,12 +428,18 @@ async def do_delete(self, id_, options):
...
 
-    @api_method(KeychainCredentialUsedByArgs, KeychainCredentialUsedByResult)
-    async def used_by(self, id_):
+    @api_method(
+        KeychainCredentialUsedByArgs,
+        KeychainCredentialUsedByResult,
+        audit="keychaincredential.used_by:",
+        audit_callback=True
+    )
+    async def used_by(self, audit_callback, id_):
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/keychain.py:437]: Why does this need to be audited?

File: src/middlewared/middlewared/plugins/keychain.py [hunk 2]
```diff
@@ -464,8 +488,12 @@ async def get_of_type(self, id_, type_):
 
             return credential
 
-    @api_method(KeychainCredentialGenerateSSHKeyPairArgs, KeychainCredentialGenerateSSHKeyPairResult,
-                roles=["KEYCHAIN_CREDENTIAL_WRITE"])
+    @api_method(
+        KeychainCredentialGenerateSSHKeyPairArgs,
+        KeychainCredentialGenerateSSHKeyPairResult,
+        roles=["KEYCHAIN_CREDENTIAL_WRITE"],
+        audit="Generate SSH Key Pair"
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/keychain.py:495]: Same here, what's interesting about generating random numbers that needs to be logged?

Comment (creatorcary) [reply src/middlewared/middlewared/plugins/keychain.py]: I agree

**Comment by creatorcary** [comment]
ping @anodos325

File: src/middlewared/middlewared/plugins/keychain.py [hunk 3]
```diff
@@ -496,8 +513,13 @@ def generate_ssh_key_pair(self):
...
         }
 
-    @api_method(KeychainCredentialRemoteSSHHostKeyScanArgs, KeychainCredentialRemoteSSHHostKeyScanResult,
-                roles=["KEYCHAIN_CREDENTIAL_WRITE"])
+    @api_method(
+        KeychainCredentialRemoteSSHHostKeyScanArgs,
+        KeychainCredentialRemoteSSHHostKeyScanResult,
+        roles=["KEYCHAIN_CREDENTIAL_WRITE"],
+        audit="Remote Host Key Scan:",
+        audit_extended=lambda data: data["host"]
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/keychain.py:521]: This also does not do any changes on the local system or expose any secrets, should it be audited?

**Review by themylogin — APPROVED** [review]

File: src/middlewared/middlewared/plugins/keychain.py (hunk 3 above)
Comment (creatorcary) [reply src/middlewared/middlewared/plugins/keychain.py:521]: Probably not

**Pull request merged**